core/src/main/java/ai/timefold/solver/core/impl/heuristic/selector/list/ElementDestinationSelector.java

```diff
@@ -97,14 +97,10 @@ public Iterator<ElementLocation> iterator() {
             if (entitySelector.getSize() == 0) {
                 return Collections.emptyIterator();
             }
-            // If the list variable allows unassigned values, add the option of unassigning.
-            var stream = listVariableDescriptor.allowsUnassignedValues() ? Stream.of(ElementLocation.unassigned())
-                    : Stream.<ElementLocation> empty();
             // Start with the first unpinned value of each entity, or zero if no pinning.
             // Entity selector is guaranteed to return only unpinned entities.
-            stream = Stream.concat(stream,
-                    StreamSupport.stream(entitySelector.spliterator(), false)
-                            .map(entity -> ElementLocation.of(entity, listVariableDescriptor.getFirstUnpinnedIndex(entity))));
+            Stream<ElementLocation> stream = StreamSupport.stream(entitySelector.spliterator(), false)
```

Comment:
Changes in this file + the test are the only relevant changes.
The rest is a bit of refactoring which I couldn't resist doing when I saw this ancient code once again.